content/custom-decorators.md

```diff
@@ -86,44 +86,44 @@ async findOne(user) {
 
 #### Passing data
 
-When the behavior of your decorator depends on some conditions, you may use the `data` param to pass an argument to the decorator's factory function. For example, the construction below:
-
-```typescript
-@@filename()
-@Get()
-async findOne(@User('test') user: UserEntity) {
-  console.log(user);
-}
-@@switch
-@Get()
-@Bind(User('test'))
-async findOne(user) {
-  console.log(user);
-}
-```
-
-Will make possible to access the `test` string via the `data` argument:
+When the behavior of your decorator depends on some conditions, you can use the `data` parameter to pass an argument to the decorator's factory function:
 
 ```typescript
 @@filename(user.decorator)
 import { createParamDecorator } from '@nestjs/common';
 
 export const User = createParamDecorator((data: string, req) => {
-  console.log(data); // test
+  req.user.userType = data;
```

Comment:
Custom decorators should be used to conditionally get the data rather than modify it on the fly. For instance:
```typescript
return data ? req.user && req.user[data] : req.user;
```
Then, someone can easily pick a single property of the user object using `@User('id')`

Reply:
Got it, that makes sense.  If I'm understanding correctly, after testing it out, this was the code sample that worked for me (slight change to yours):

```return data && req.user && req.user[data] ? req.user[data] : req.user;```

Does that make sense?  I'm returning the property based on the key passed to the decorator if it exists, otherwise the regular req.user, right?

This ability to pick a property via a decorator with passed key seems like a major point to get across (it eluded me until this discussion),  so if I'm on the right track, once this code sample is good, I'll make sure this point is clear.

Reply:
> Does that make sense? I'm returning the property based on the key passed to the decorator if it exists, otherwise the regular req.user, right?

It seems to introduce side-effects. I think that this:

```typescript
return data ? req.user && req.user[data] : req.user;
```
should be enough (just return `undefined` if the property is `undefined` 😄 )